themes/default/content/resources/walkthrough-of-building-container-based-aws-lambda-functions/index.md

```diff
@@ -0,0 +1,75 @@
+
+    All it's doing is providing some additional information about what's actually being processed. And again, we give it a policy. So that's, that's pretty much it here. And you can see this is, this is looking for that JPEG thumbnail that gets written once it gets written. And so, nothing too crazy here. So instead of a role, I just give it a policy because this actually creates a role on its own as well. So that's kind of how all that works. And so really the only code left look at is how do we build that container image and what is the code running within that container image? So if we go over here, we can look at our Docker file and you can see up here, I'm basing this image off of the image that AWS has published.
+
+    This is actually using their new public ECR. And so we're grabbing that image from there. And then we're just adding a few things that weren't present in the image already. So we're including a few utilities. We need those to install a couple other things. So we need unzip to install the AWS CLI where we just grabbed this CLI and then we unzip it and install it. And then we similarly need 'tar xz' to install FFmpeg. So we also grab that file unpack it, and then move that binary into somewhere on the path. And then once all that's done you know, finally we just copy this handler and then set the entry point to be this this handler. And so if we look at this handler it's fairly simple.
+
+    This is exactly the same code you would write in a Lambda function. And so here, all we're doing is we have this, you simple run wrapper that does a 'exec'. And so that's simply a shelling out to take the file that came in and copy it to the local disk. We run FFmpeg against it with the correct FFmpeg parameters to grab the thumbnail that outputs to the thumbnail. And then we finally copy that thumbnail that we create back into S3. And so that's how all this is wired together fairly simple and straightforward. And so that really that's how the entire example works, and like I said in the short video, it really is just a few dozen lines of code to wire, everything together pulling me makes it super simple and easy to get started doing this. So yeah, I hope you enjoy this video. Be really curious to see what you do with the Lambda functions using container-based functions. And I hope to see you next week on Modern Infrastructure Wednesday. If you liked this video, please leave a comment below. Would love to hear what other content you would like to see as well as please make sure you hit that like and subscribe button so we know you're enjoying the content here. Thanks.
```

Comment:
```suggestion
    This is exactly the same code you would write in a Lambda function. And so here, all we're doing is we have this simple run wrapper that does an 'exec'. And so that's simply a shelling out to take the file that came in and copy it to the local disk. We run FFmpeg against it with the correct FFmpeg parameters to grab the thumbnail that outputs to the thumbnail. And then we finally copy that thumbnail that we create back into S3. And so that's how all this is wired together fairly simple and straightforward. And so that really that's how the entire example works, and like I said in the short video, it really is just a few dozen lines of code to wire everything together pulling me makes it super simple and easy to get started doing this. So yeah, I hope you enjoy this video. Be really curious to see what you do with the Lambda functions using container-based functions. And I hope to see you next week on Modern Infrastructure Wednesday. If you liked this video, please leave a comment below. Would love to hear what other content you would like to see as well as please make sure you hit that like and subscribe button so we know you're enjoying the content here. Thanks.
```